src/app/submission/pages/submission.component.ts

```diff
@@ -277,18 +277,21 @@ export class SubmissionComponent implements OnInit, OnDestroy {
     this.downloadDisabled = true;
     const uuid = this.submissionEnvelope['uuid']['uuid'];
     this.loaderService.display(true, 'This may take a moment. Please wait...');
+    // now set a cookie to disable further download attempt within X min of this request or
+    // until a response is returned when the cookie gets deleted.
+    this.cookieService.set(this.DOWNLOAD_TIMEOUT_COOKIE, "1", {expires: this.DOWNLOAD_BACKOFF_MINS / (24*60)})
```

Comment:
Instead of using a comment to specify intent, a properly named function can be used.